addons/account/models/account_move.py

```diff
@@ -466,6 +466,14 @@ def _get_counterpart(self):
         ('credit_debit2', 'CHECK (credit+debit>=0)', 'Wrong credit or debit value in accounting entry !'),
     ]
 
+    @api.constrains('analytic_account_id', 'name')
+    def _check_analytic_account(self):
+        for line in self:
+            if line.analytic_account_id and not line.name:
+                raise ValidationError(_('''You should put a label on this account item if you are to use an analytic account.\n
+                                        Account: %s %s, Debit: %s, Credit: %s''') %
+                                        (line.account_id.code, line.account_id.name, line.debit, line.credit))
```

Comment:
I'd rather simply go for:
"Please define a label on all the journal items using an analytic account."